src/engine/resolvers.rs

```diff
@@ -80,11 +82,60 @@ where
         }
     }
 
+    /// Returns a neo4j database driver from the pool.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error`] variant [`DatabaseMismatch`] if the feature `neo4j`
+    /// is not enabled.
+    ///
+    /// Returns an [`Error]` variant [`Neo4jPoolGetConnectionFailed`] if a new
+    /// connection to the database cannot be retrieved from the pool.
+    ///
+    /// [`Error`]: ../../error/enum.Error.html
+    /// [`DatabaseMismatch`]: ../../error/enum.Error.html#variant.DatabaseMismatch
+    /// [`Neo4jPoolGetConnectionFailed`]: ../../error/enum.Error.html#variant.Neo4jPoolGetConnectionFailed
+    ///
+    /// # Examples
+    ///
+    /// ```rust, no_run
+    /// # use tokio::runtime::Runtime;
+    /// # use warpgrapher::engine::resolvers::{ResolverFacade, ExecutionResult};
+    ///
+    /// fn custom_resolve(facade: ResolverFacade<(), ()>) -> ExecutionResult {
+    ///     let mut rt = Runtime::new().unwrap();
+    ///     rt.block_on(async {
+    ///
+    ///         let neo4j_client = facade.db_as_neo4j().await.unwrap();
+    ///         // use neo4j client
+    ///
+    ///     });
+    ///     facade.resolve_null()
+    /// }
+    /// ```
+    #[cfg(feature = "neo4j")]
+    pub async fn db_as_neo4j(
+        &self,
+    ) -> Result<bb8::PooledConnection<'_, bb8_bolt::BoltConnectionManager>, Error> {
+        let db_pool: &bb8::Pool<bb8_bolt::BoltConnectionManager> =
+            match self.executor().context().pool() {
+                DatabasePool::Neo4j(p) => p,
+                _ => {
+                    return Err(Error::DatabaseMismatch {});
+                }
+            };
+        let neo4j_client = db_pool
+            .get()
```

Comment:
Given this call, this may not be appropriate for the naming convention of as_neo4j().  It's probably more appropriate for it to be into_neo4j(), given that it's making a call to a call with variable expense to a non-copy operation in order to produce the result.

Reply:
No objection.

Reply:
Fixed.